src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -91,10 +91,12 @@ public void RunApiCompat()
                             IsBaselineSuppression = _isBaselineSuppression
                         };
 
+                        // If the error is suppressed, don't log anything.
+                        if (_log.SuppressionEngine.IsErrorSuppressed(suppression))
```

Comment:
Does this work in the incremental case? AFAIR, we are rewriting the whole suppression file if it already existed everytime that you run this, so the reason why we would still log an error even when the error was already suppressed, was so that when the suppression file got overridden it would contain the error that was previously suppressed.

Reply:
Yes it does. The logger implementations immediately return if the error is suppressed: https://github.com/dotnet/sdk/blob/cf5ff15fd295f02cef652cf5e9a1c9879a0615bb/src/Compatibility/Microsoft.DotNet.Compatibility.Console/ConsoleCompatibilityLogger.cs#L26-L27. I didn't change that with this or my previous change.

Correct. The suppression file is (re-)written if the `GenerateCompatibilitySuppressionFile` setting is true and if a suppression file path is passed in. Its previous content is ignored as it's always regenerated.

If `IsErrorSuppressed` returns true, then that means that the error/suppression is either already part of the suppression backing storage or that a `NoWarn` setting was passed in to ignore the error. The `WriteSuppressionsToFile` serializes the backing storage to an xml file.

I do think this works as intended and as mentioned, this change shouldn't mutate the existing behavior. This makes sure that. This change avoids the LogError call by frontloading the IsErrorSuppressed check that already exists for the header logging condition.